### PR TITLE
Add `ruby` support with `ruby` tests

### PR DESCRIPTION
**Pull request by harshsharma2002**
Added ruby support along with tests for ruby.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,13 +2,14 @@ FROM docker.io/library/node:16.17.0-alpine3.16
...
-    apk add --no-cache gcc g++ musl-dev python3 openjdk17 iptables ip6tables
+    apk add --no-cache gcc g++ musl-dev python3 openjdk17 ruby iptables ip6tables
 
 RUN set -ex && \
     rm -f /usr/libexec/gcc/x86_64-alpine-linux-musl/6.4.0/cc1obj && \
     rm -f /usr/libexec/gcc/x86_64-alpine-linux-musl/6.4.0/lto1 && \
     rm -f /usr/libexec/gcc/x86_64-alpine-linux-musl/6.4.0/lto-wrapper && \
-    rm -f /usr/bin/x86_64-alpine-linux-musl-gcj
+    rm -f /usr/bin/x86_64-alpine-linux-musl-gcj && \
+    rm -rf /var/cache/apk/*
```
Comment (anipnwr7777) [inline Dockerfile:12]: any specific reason for this cleanup ?

Comment (harshsharma2002) [reply Dockerfile:12]: it just reduces the size of the image, not really important to have, can be removed.

Comment (anipnwr7777) [reply Dockerfile:12]: do you by any chance know how much benefit in terms of size of image are we getting by adding this line ?

Comment (harshsharma2002) [reply Dockerfile:12]: ok not much, it is 2 MB

Comment (anipnwr7777) [reply Dockerfile:12]: Ok NP. For now let's go without this, we can take that up in future if needed.

**Review by anipnwr7777 — APPROVED** [review]
LG

**Pull request merged**